packages/flutter_tools/lib/src/commands/flutter_command_runner.dart

```diff
@@ -108,39 +134,47 @@ class FlutterCommandRunner extends CommandRunner {
 
     return super.runCommand(globalResults);
   }
 
   List<BuildConfiguration> _createBuildConfigurations(ArgResults globalResults) {
-    if (!FileSystemEntity.isDirectorySync(ArtifactStore.packageRoot)) {
-      String message = '${ArtifactStore.packageRoot} is not a valid directory.';
-      if (ArtifactStore.packageRoot == 'packages') {
-        if (FileSystemEntity.isFileSync('pubspec.yaml'))
-          message += '\nDid you run `pub get` in this directory?';
-        else
-          message += '\nDid you run this command from the same directory as your pubspec.yaml file?';
-      }
-      stderr.writeln(message);
-      throw new ProcessExit(2);
-    }
-
-    String enginePath = globalResults['engine-src-path'];
+    String enginePath = globalResults['engine-src-path'] ?? Platform.environment[kFlutterEngineEnvironmentVariableName];
     bool isDebug = globalResults['debug'];
     bool isRelease = globalResults['release'];
     HostPlatform hostPlatform = getCurrentHostPlatform();
+    TargetPlatform hostPlatformAsTarget = getCurrentHostPlatformAsTarget();
 
-    if (enginePath == null && globalResults['local-build']) {
-      Directory flutterDir = new Directory(path.join(globalResults['package-root'], 'flutter'));
-      String realFlutterPath = flutterDir.resolveSymbolicLinksSync();
-
-      enginePath = path.dirname(path.dirname(path.dirname(path.dirname(realFlutterPath))));
-      bool dirExists = FileSystemEntity.isDirectorySync(path.join(enginePath, 'out'));
-      if (enginePath == '/' || enginePath.isEmpty || !dirExists) {
-        stderr.writeln('Unable to detect local build in $enginePath.\n'
-            'Do you have a dependency override for the flutter package?');
+    if (enginePath == null && (isDebug || isRelease)) {
+      if (ArtifactStore.isPackageRootValid) {
+        Directory engineDir = new Directory(path.join(ArtifactStore.packageRoot, kFlutterEnginePackageName));
+        try {
+          String realEnginePath = engineDir.resolveSymbolicLinksSync();
+          enginePath = path.dirname(path.dirname(path.dirname(path.dirname(realEnginePath))));
+          bool dirExists = FileSystemEntity.isDirectorySync(path.join(enginePath, 'out'));
+          if (enginePath == '/' || enginePath.isEmpty || !dirExists)
+            enginePath = null;
+        } on FileSystemException { }
+      }
+      if (enginePath == null) {
+        String tryEnginePath(String enginePath) {
+          if (FileSystemEntity.isDirectorySync(path.join(enginePath, 'out')))
+            return enginePath;
+          return null;
+        }
+        enginePath = tryEnginePath(path.join(ArtifactStore.flutterRoot, '../$kFlutterEnginePackageName/src')) ??
+                     tryEnginePath(path.join(ArtifactStore.flutterRoot, '../engine/src')) ??
+                     tryEnginePath(path.join(ArtifactStore.flutterRoot, '../sky_engine/src')) ??
+                     tryEnginePath(path.join(ArtifactStore.flutterRoot, '../flutter_engine/src'));
```

Comment:
The golden path in the rest of the docs is actually the second one (matching the repository name).  The first one has the name "sky" in it, which we'll presumably want to remove eventually.